src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php

```diff
@@ -268,30 +267,30 @@ protected function assertValidMappingConfiguration(array $mappingConfig, $object
      */
     protected function detectMetadataDriver($dir, ContainerBuilder $container)
     {
-        // add the closest existing directory as a resource
         $configPath = $this->getMappingResourceConfigDirectory();
-        $resource = $dir.'/'.$configPath;
-        while (!is_dir($resource)) {
-            $resource = dirname($resource);
-        }
+        $extension = $this->getMappingResourceExtension();
 
-        $container->addResource(new FileResource($resource));
+        if (glob($dir.'/'.$configPath.'/*.'.$extension.'.xml')) {
+            $driver = 'xml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.yml')) {
+            $driver = 'yml';
+        } elseif (glob($dir.'/'.$configPath.'/*.'.$extension.'.php')) {
+            $driver = 'php';
+        } else {
+            // add the directory itself as a resource
+            $container->fileExists($dir);
```

Comment:
I guess the code is as it is because `$configPath` can be a glob, so the loop just removes "glob-dirs" until a real one is found.